modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java

```diff
@@ -44,7 +48,36 @@ public static String getDBType(String dbType, String type) {
 		return dbType;
 	}
 
+	public static long getListTypeDefinitionId(
+		long companyId,
+		ListTypeDefinitionLocalService listTypeDefinitionLocalService,
+		ObjectField objectField) {
+
+		if (!StringUtil.equals(
+				objectField.getBusinessTypeAsString(),
+				ObjectFieldConstants.BUSINESS_TYPE_PICKLIST)) {
+
+			return 0L;
+		}
+
+		long listTypeDefinitionId = GetterUtil.getLong(
+			objectField.getListTypeDefinitionId());
+
+		if (listTypeDefinitionId != 0L) {
+			return objectField.getListTypeDefinitionId();
+		}
+
+		ListTypeDefinition listTypeDefinition =
+			listTypeDefinitionLocalService.
+				fetchListTypeDefinitionByExternalReferenceCode(
+					companyId,
+					objectField.getListTypeDefinitionExternalReferenceCode());
+
+		return listTypeDefinition.getListTypeDefinitionId();
```

Comment:
@murilostodolni should we check if listTypeDefinition is null?

Reply:
@carolmariaabb it's make sense, if is null we throw `ObjectFieldListTypeException`?